dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -588,4 +621,78 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
                 "Removing bundles in a separated process, the result of the operation will be notified")).build();
     } // deleteAllSuccess.
 
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.TEXT_PLAIN})
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public final Response uploadBundle(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+            FormDataMultiPart multipart,
+            @Context UriInfo uriInfo, @QueryParam("sync") boolean sync) throws DotPublisherException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        for (final BodyPart part : multipart.getBodyParts()) {
+
+            try(InputStream inputStream = part.getEntity() instanceof InputStream ? (InputStream) part
+                    .getEntity()
+                    : Try.of(() -> part.getEntityAs(InputStream.class)).getOrNull()) {
+
+                if (inputStream == null) {
+                    continue;
+                }
+                final ContentDisposition meta = part.getContentDisposition();
+                if (meta == null) {
+                    continue;
+                }
+                final String fileName = meta.getFileName();
+                if (UtilMethods.isNotSet(fileName) || fileName.startsWith(".") || fileName
```

Comment:
Codacy found an issue: [This call to String.startsWith can be rewritten using String.charAt(0)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5747250)